src/result/position.rs

```diff
@@ -0,0 +1,73 @@
+use std::fmt::{Display, Error};
+
+/// Position represents the location within an Ion stream where an error has been
+/// identified. For all formats `byte_offset` will contain the number of bytes into the stream
+/// that have been processed prior to encountering the error. When working with the text format,
+/// `line_column` will be updated to contain the line and column as well.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Position {
+    pub(crate) byte_offset: usize,
+    pub(crate) line_column: Option<(usize, usize)>,
+}
+
+impl Position {
+    /// Creates a new Position with the provided offset in bytes.
+    /// Line and Column offsets can be added using [`Self::with_line_and_column()`].
+    pub fn with_offset(offset: usize) -> Self {
+        Position {
+            byte_offset: offset,
+            line_column: None,
+        }
+    }
+
+    /// Add line and column information to the current Position.
+    pub fn with_line_and_column(&self, line: usize, column: usize) -> Self {
+        Position {
+            line_column: Some((line, column)),
+            ..*self
+        }
+    }
+
+    /// Returns the offset from the start of the Ion stream in bytes.
+    pub fn byte_offset(&self) -> usize {
+        self.byte_offset
+    }
+
+    /// If available returns the text position as line and column offsets.
+    pub fn line_and_column(&self) -> Option<(usize, usize)> {
```

Comment:
🗺️ This accessor was previously called `text_position()`. I renamed it to `line_and_column()` to be consistent with the variable name and to make it more obvious what the pair of values being returned each represented.